### PR TITLE
refactor(Move): Create abstract class MenuFoldersFragment

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on #552

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt [hunk 1]
```diff
@@ -40,22 +39,24 @@ import com.infomaniak.mail.BuildConfig
...
 
-class MenuDrawerFragment : Fragment() {
+class MenuDrawerFragment : MenuFoldersFragment() {
 
     private lateinit var binding: FragmentMenuDrawerBinding
-    private val mainViewModel: MainViewModel by activityViewModels()
+
+    override val inboxFolder: MenuDrawerItemView by lazy { binding.inboxFolder }
+    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
+    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt:55]:
```suggestion
    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
```

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt [hunk 2]
```diff
@@ -87,15 +86,18 @@ class MenuDrawerFragment : Fragment() {
         observeQuotas()
     }
 
+    override fun onFolderClicked(folderId: String) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt:89]: Rename all `...Clicked` stuff to `...Selected`.

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt [hunk 3]
```diff
@@ -198,7 +199,7 @@ class MenuDrawerFragment : Fragment() {
 
     private fun observeCurrentFolder() {
         mainViewModel.currentFolder.observe(viewLifecycleOwner) { folder ->
-            binding.inboxFolder.setSelectedState(folder.role == FolderRole.INBOX)
+            binding.inboxFolder.setSelectedState(folder.role == Folder.FolderRole.INBOX)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt:202]: Revert this change.

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.views.MenuDrawerItemView
+
+abstract class MenuFoldersFragment : Fragment() {
+
+    protected val mainViewModel: MainViewModel by activityViewModels()
+
+    protected abstract val customFoldersList: RecyclerView
+    protected abstract val defaultFoldersList: RecyclerView
+    protected abstract val inboxFolder: MenuDrawerItemView
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt:34]:
```suggestion
    protected abstract val inboxFolder: MenuDrawerItemView
    protected abstract val defaultFoldersList: RecyclerView
    protected abstract val customFoldersList: RecyclerView
```

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    protected var inboxFolderId: String? = null
+
+    protected val customFoldersAdapter: FolderAdapter by lazy {
+        FolderAdapter(onClick = ::onFolderClicked, isInMenuDrawer = isInMenuDrawer)
+    }
+
+    protected val defaultFoldersAdapter: FolderAdapter by lazy {
+        FolderAdapter(onClick = ::onFolderClicked, isInMenuDrawer = isInMenuDrawer)
+    }
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt:47]:
```suggestion
    protected val defaultFoldersAdapter: FolderAdapter by lazy {
        FolderAdapter(onClick = ::onFolderClicked, isInMenuDrawer = isInMenuDrawer)
    }

    protected val customFoldersAdapter: FolderAdapter by lazy {
        FolderAdapter(onClick = ::onFolderClicked, isInMenuDrawer = isInMenuDrawer)
    }

```

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    private fun setupAdapters() {
+        defaultFoldersList.adapter = defaultFoldersAdapter
+        customFoldersList.adapter = customFoldersAdapter
+    }
+
+    private fun setupListeners() {
+        inboxFolder.setOnClickListener { inboxFolderId?.let(::onFolderClicked) }
+    }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuFoldersFragment.kt:64]:
```suggestion
        setupListeners()
        setupAdapters()
    }

    private fun setupListeners() {
        inboxFolder.setOnClickListener { inboxFolderId?.let(::onFolderClicked) }
    }

    private fun setupAdapters() {
        defaultFoldersList.adapter = defaultFoldersAdapter
        customFoldersList.adapter = customFoldersAdapter
    }
```

File: app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt [hunk 7]
```diff
@@ -21,32 +21,25 @@ import android.os.Bundle
...
 
     private lateinit var binding: FragmentMoveBinding
     private val navigationArgs: MoveFragmentArgs by navArgs()
-    private val mainViewModel: MainViewModel by activityViewModels()
     private val moveViewModel: MoveViewModel by viewModels()
 
-    private var inboxFolderId: String? = null
+    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
+    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
+    override val inboxFolder: MenuDrawerItemView by lazy { binding.inboxFolder }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt:40]:
```suggestion
    override val inboxFolder: MenuDrawerItemView by lazy { binding.inboxFolder }
    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
```

**Review by KevinBoulongne — APPROVED** [review]

**Review by JorisBodin — APPROVED** [review]
